packages/ui/src/components/templates/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {Dimensions, StyleSheet, View} from 'react-native';
+import {
+  BarChartDashboard,
+  LineChartDashboard,
+  PieChartDashboard,
+} from '../../molecules';
+import {ScrollView} from '../../atoms';
+
+interface Data {
+  value: number;
+  color?: string;
+  label: string;
+}
+
+interface Graph {
+  type: 'pie' | 'bar' | 'line';
+  dataList: Data[][];
+  title?: string;
+}
+
+interface Line {
+  graphList: Graph[];
+}
+
+interface DashboardProps {
+  style?: any;
+  lineList: Line[];
+}
+
+const styleGraph = (nbGraphInLine: number, type: 'style' | 'width') => {
+  let style = null;
+  let width = null;
+
+  if (nbGraphInLine === 1) {
+    style = {width: Dimensions.get('window').width - 60};
+    width = Dimensions.get('window').width - 160;
+  } else if (Dimensions.get('window').width < 500 && nbGraphInLine !== 1) {
+    style = {width: Dimensions.get('window').width / 2 - 15};
+    width = Dimensions.get('window').width / 4;
+  } else if (nbGraphInLine === 2) {
+    style = {width: Dimensions.get('window').width / 2 - 60};
+    width = Dimensions.get('window').width / 3 - 60;
+  } else if (nbGraphInLine === 3) {
+    style = {width: Dimensions.get('window').width / 3 - 60};
+    width = Dimensions.get('window').width / 5 - 60;
+  } else if (nbGraphInLine === 4) {
+    style = {width: Dimensions.get('window').width / 4 - 20};
+    width = Dimensions.get('window').width / 6 - 60;
+  }
+
+  return type === 'style' ? style : width;
```

Comment:
This can be simplify, instead of using type ('style' | 'width'), return both values as a object and redistribute them to the right props.
If you can find a formula depending on index, it would be great too but if not that's not really a problem